### ISSUE TITLE
JAX vmap implementation

### ISSUE THREAD

**Issue by umangjpatel** [umangjpatel/dnet#7]
**Description**
Implement JAX vmap structure for batching

**Solution**
Use JAX vmap for batched gradient descent implementation.

**Additional context**
Refer to https://jax.readthedocs.io/en/latest/index.html
